lib/node_modules/@stdlib/blas/ext/join/lib/base.js

```diff
@@ -0,0 +1,99 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var dtypes = require( '@stdlib/ndarray/dtypes' );
+var gjoin = require( '@stdlib/blas/ext/base/ndarray/gjoin' );
+var factory = require( '@stdlib/ndarray/base/unary-reduce-strided1d-dispatch-factory' );
+
+
+// VARIABLES //
+
+var idtypes0 = dtypes( 'all' ); // input ndarray
+var idtypes1 = dtypes( 'all' ); // separator ndarray
+var odtypes = dtypes( 'all' );
+var policies = {
+	'output': 'same',
+	'casting': 'promoted'
+};
```

Comment:
```suggestion
var idtypes0 = dtypes( 'all' ); // input ndarray
var idtypes1 = [ 'generic' ];   // separator ndarray
var odtypes = [ 'generic' ];
var policies = {
	'output': 'same', // note: because we always return a "generic" ndarray, this policy is effectively ignored
	'casting': 'none'
};
```